src/libsyntax/feature_gate.rs

```diff
@@ -206,9 +213,12 @@ declare_features! (
     // #23121. Array patterns have some hazards yet.
     (active, slice_patterns, "1.0.0", Some(23121), None),
 
-    // Allows the definition of `const fn` functions.
+    // Allows the definition of `const fn` functions with some advanced features
     (active, const_fn, "1.2.0", Some(24111), None),
```

Comment:
Well, we will want it for `NonZero::new_unchecked`, right?

Reply:
Hmm... I was hoping on using the stabilization of `min_const_fn` but exclusion of `const_fn` from libstd so that standard library functions could be `const`-ified en masse instead of piecemeal. This kinda relies on not having `#![feature(const_fn)]` enabled so that accidental things don't happen.

Could we handle `new_unchecked` through some other internal means instead temporarily?

Reply:
I'd say:

* `const fn` without `#[rustc_const_unstable]` is stable if it passes `min_const_fn`
* `const fn` not passing `min_const_fn` can be forced stable with `#[rustc_const_fn_force_stable]`

This way the active feature gates have no influence on stability, but we're checking the right thing and can't accidentally do anything wrong.

Reply:
@oli-obk  this sounds good to me :)